lib/perfdata/elasticsearchwriter.cpp

```diff
@@ -410,6 +427,76 @@ void ElasticsearchWriter::Enqueue(const Checkable::Ptr& checkable, const String&
 	}
 }
 
+void ElasticsearchWriter::DiscoverVersion()
+{
+	namespace http = boost::beast::http;
+
+	OptionalTlsStream stream;
+
+	try {
+		stream = Connect();
+	} catch (const std::exception& ex) {
+		Log(LogWarning, "ElasticsearchWriter")
+			<< "Discover failed (assuming version " << GetDiscoveredVersion()
+			<< "), cannot connect to Elasticsearch: " << DiagnosticInformation(ex, false);
+		return;
```

Comment:
So if anything in this function fails, it just continues and assumes the old version? Why? I mean if this request failed, the server is probably down and might come back at a later time and then it will receive requests in the wrong format? Why isn't the request retried? I mean if it fails, the requests sending the actual data will probably fail as well.

Reply:
Improved it.

```
[2023-06-30 12:19:59 +0200] warning/ElasticsearchWriter: Failed to parse HTTP response from host '127.0.0.1' port '9200': Error: end of stream
[2023-06-30 12:20:01 +0200] notice/Process: Running command '/Users/aklimov/NET/WS/prefix/usr/lib/nagios/plugins/check_ping' '-6' '-H' '::1' '-c' '200,15%' '-w' '100,5%': PID 56170
[2023-06-30 12:20:01 +0200] notice/Process: PID 56170 ('/Users/aklimov/NET/WS/prefix/usr/lib/nagios/plugins/check_ping' '-6' '-H' '::1' '-c' '200,15%' '-w' '100,5%') terminated with exit code 128
[2023-06-30 12:20:01 +0200] warning/PluginCheckTask: Check command for object 'Alexanders-MacBook-Pro-2.local!ping6' (PID: 56170, arguments: '/Users/aklimov/NET/WS/prefix/usr/lib/nagios/plugins/check_ping' '-6' '-H' '::1' '-c' '200,15%' '-w' '100,5%') terminated with exit code 128, output: execvpe(/Users/aklimov/NET/WS/prefix/usr/lib/nagios/plugins/check_ping) failed: No such file or directory
[2023-06-30 12:20:02 +0200] notice/ElasticsearchWriter: Connecting to Elasticsearch on host '127.0.0.1' port '9200'.
[2023-06-30 12:20:03 +0200] notice/ElasticsearchWriter: Backend identified itself as v5
```